src/lib/server-api/server-api.ts

```diff
@@ -0,0 +1,191 @@
+        });
+    } 
+
+    function typecheckFile(path: string) {
+        const msg = {
+            typehint: "TypecheckFileReq",
+            fileInfo: {
+                file: path
+            }
+        }
+        return client.post(msg);
+    } 
+
+    function symbolByName(qualifiedName) {
+        const msg = {
+            typehint: 'SymbolByNameReq',
+            typeFullName: qualifiedName
+        }
+        return client.post(msg);
+    }
+        
+    function formatSourceFile(path, contents, callback) {
+        return withTempFile(path, contents).then((tempFilePath) => {
+            const req = {
+                typehint: "FormatOneSourceReq",
+                file: {
+                    file: path,
+                    contentsIn: tempFilePath
+                }
+            }
+            return client.post(req);
+        });
+    }
+
+
+    function getImplicitInfo(path: string, startO: number, endO: number) {
+        const msg = {
+            "typehint":"ImplicitInfoReq",
+            "file": path,
+            "range": {
+                "from": startO,
+                "to": endO
+            }
+        }
+        return client.post(msg)
+    }
+
+
+    function typecheckAll() {
+        client.post({"typehint": "TypecheckAllReq"});
+    }
+
+    function unloadAll() {
+        client.post({"typehint": "UnloadAllReq"});
+    }
+
+    function getRefactoringPatch(procId: number, refactoring: RefactoringDesc) {
+        const req = {
+            typehint: 'RefactorReq',
+            procId: procId,
+            params: refactoring,
+            interactive: false
+        }
+        return client.post(req);
+    }
+
+    function searchPublicSymbols(keywords: string[], maxSymbols: number) {
+        return client.post({
+            typehint: "PublicSymbolSearchReq",
+            keywords: keywords,
+            maxResults: maxSymbols
+        })
+    }
+
+    
+
+    function getDocUriAtPoint(file: string, point: Point) {
+        return client.post({
+            typehint: "DocUriAtPointReq",
+            file: file,
+            point: point
+        });
+    }
+
+    return {
+        getCompletions,
+        getSymbolAtPoint,
+        typecheckFile,
+        typecheckBuffer,
+        symbolByName,
+        formatSourceFile,
+        getImplicitInfo,
+        typecheckAll,
+        unloadAll,
+        getRefactoringPatch,
+        searchPublicSymbols,
+        getDocUriAtPoint
+    }
+}
+
+
+
+export interface Api {
+    getCompletions: (filePath: string, bufferText: any, offset: any, noOfAutocompleteSuggestions: any) => Promise<Typehinted>;
+    getSymbolAtPoint: (path: string, offset: any) => Promise<Typehinted>;
+    typecheckFile: (path: string) => Promise<Typehinted>;
+    typecheckBuffer: (path: string, text: string) => void;
+    symbolByName: (qualifiedName: any) => Promise<Typehinted>;
+    formatSourceFile: (path: any, contents: any, callback: any) => Promise<Typehinted>;
```

Comment:
Definitely! It's a work in progress. It's just ensime-atom using it now, but as soon as vscode gets going it will be natural to actually create the API types and refine a lot. I'm just not trying to batch any work until I need it… Trying to be agile :)